### PR TITLE
split airbyte protocol generated structs out of base_python

### PR DESCRIPTION
**Pull request by cgardens**
## What
* See issue for more color: https://github.com/airbytehq/airbyte/issues/718
* Splits the generated airbyte protocol structs into their own lib so that they can be depended upon without needing to depend on base_python.
* after this PR:
  * `airbyte-protocol` - lib with structs
  * `base-python` - the base python stuff
* try to get all of the `dependsOn` right in gradle for the python modules. in some cases we were missing `dependsOn` statements leading to non-deterministic behavior. honestly not sure that i got them all yet.

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres-singer/base_singer [hunk 1]
```diff
@@ -1 +0,0 @@
-../../bases/base-singer/base_singer
```
[the remark this answers is not in the document]
Comment (ChristopheDuong) [reply airbyte-integrations/connectors/source-postgres-singer/base_singer:1]: It was on purpose for DX: https://github.com/airbytehq/airbyte/pull/716#discussion_r513275905

**Review by ChristopheDuong — APPROVED** [review]

File: airbyte-integrations/bases/airbyte-protocol/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+airbyte_protocol/models/yaml
```
Comment (jrhizor) [inline airbyte-integrations/bases/airbyte-protocol/.gitignore:1]: There isn't anything in this location. This `.gitignore` file can be deleted.

File: airbyte-integrations/bases/airbyte-protocol/main_dev.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from airbyte_protocol.entrypoint import main
```
Comment (jrhizor) [inline airbyte-integrations/bases/airbyte-protocol/main_dev.py:25]: This is in the base, not `airbyte-protocol`? Can we just delete this file? I don't think we need it for a library class.

**Review by jrhizor — APPROVED** [review]

File: airbyte-integrations/bases/airbyte-protocol/main_dev.py (hunk 3 above)
Comment (cgardens) [reply airbyte-integrations/bases/airbyte-protocol/main_dev.py:25]: agreed.

File: airbyte-integrations/bases/airbyte-protocol/.gitignore (hunk 2 above)
Comment (cgardens) [reply airbyte-integrations/bases/airbyte-protocol/.gitignore:1]: thanks.

**Review by michel-tricot — COMMENTED** [review]
great!

**Pull request merged**